Выполнил ФЗ 15

Kvazzimoda opened the pull request.

Kvazzimoda committed:
Выполнил ФЗ 15

Kvazzimoda committed:
Добавил файл docker-compose.yml для прохождения тестов

Kvazzimoda committed:
Очередная попытка победить тесты

Kvazzimoda committed:
Очередная попытка победить тесты

Kvazzimoda committed:
Очередная попытка победить тесты

Kvazzimoda committed:
Очередная попытка победить тесты

Kvazzimoda committed:
Очередная попытка победить тесты

Kvazzimoda committed:
Очередная попытка победить тесты

Kvazzimoda committed:
Попытка победить тест из-за валидации дат

mikita95 commented on src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java:
```diff
@@ -46,65 +57,88 @@ public BookingDto createBooking(Long userId, BookingDto bookingDto) {
+                    .filter(b -> b.getStart().isAfter(now))
+                    .map(BookingMapper::toDto)
+                    .collect(Collectors.toList());
+            case "WAITING" -> bookings.stream()
```
Лучше было бы завести enum - так код было бы проще поддерживать и меньше вероятность ошибки

mikita95 commented on src/main/java/ru/practicum/shareit/item/comment/Comment.java:
```diff
@@ -0,0 +1,31 @@
+
+@Entity
+@Table(name = "comments")
+@Data
```
Лучше не использовать аннотацию Data над сущностями jpa, так как она генерирует методы hashCode и equals, что может привести к инициализации lazy полей или циклическим вызовам

mikita95 commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,43 @@
+    name VARCHAR(255) NOT NULL,
+    description VARCHAR(1024) NOT NULL,
+    available BOOLEAN NOT NULL,
+    owner_id BIGINT NOT NULL,
```
Стоит добавить индексы на колонки, по которым часто выполняются запросы, так будет эффективнее при большом количестве данных

mikita95 reviewed (changes requested):
Все сделано хорошо :). Оставил пару замечаний

Kvazzimoda committed:
1.Завёл enum для статуса бронирования
2.Убрал аннотацию Data в сущностях JPA
3.Добавил индексы на колонки схемы, по котором часто происходят запросы

mikita95 reviewed (approved):
Все отлично :) Можно мержить пул реквест
